File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    total = len(priorities)
    answer = 0
    cur = 0
    while True:
        if max(priorities) == priorities[cur % total]:
            priorities[cur % total] = 0
            answer += 1
            if cur % total == location: break
        cur += 1
        
    return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    let arr = priorities.map((v, i) => {
        return {val: v, idx: i};
    });

    let answer = [];

    while (arr.length > 0) {
        let cur = arr.shift();
        let notMax = arr.some(v => v.val > cur.val);
        if (notMax) arr.push(cur);
        else answer.push(cur);
    }

    return answer.findIndex(v => v.idx === location) + 1;
}